File: fix_getUserName.js
const fs = require('fs');

function fix(file, usersVar) {
  if (!fs.existsSync(file)) return;
  let code = fs.readFileSync(file, 'utf8');
  
  // Replace standard block
  const pattern = new RegExp(`const getUserName = \\(serviceNumber\\) => \\{[\\s\\S]*?return[^;]+;\\s*\\};`, 'g');
  
  const replacement = `const getUserName = (serviceNumber) => {
    if (!serviceNumber || String(serviceNumber).trim() === '') return 'Unassigned';
    if (!Array.isArray(${usersVar})) return serviceNumber;
    const foundUser = ${usersVar}.find(
      (user) => String(user.service_number) === String(serviceNumber) || String(user.serviceNum) === String(serviceNumber)
    );
    return foundUser ? (foundUser.display_name || foundUser.user_name || foundUser.name || serviceNumber) : serviceNumber;
  };`;
  
  const newCode = code.replace(pattern, replacement);
  fs.writeFileSync(file, newCode);
  console.log('Fixed', file);
}

fix('src/pages/Admin/AdminMyTeamIncidentViewAll/AdminMyTeamIncidentViewAll.jsx', 'users');
fix('src/pages/Admin/AdminAllIncidents/AdminAllIncidents.jsx', 'users');
fix('src/pages/SuperAdmin/SuperAdminAllIncident/SuperAdminAllIncident.jsx', '(users || [])');
fix('src/pages/User/UserMyTeamIncidentViewAll/UserMyTeamIncidentViewAll.jsx', 'users');
fix('src/pages/Technician/TechnicianAllTeam/TechnicianAllTeam.jsx', 'allUsers');
fix('src/components/IncidentHistory/IncidentHistory.jsx', 'users');

File: fix_getUserName.py
import os, re

files = {
    'src/pages/Admin/AdminMyTeamIncidentViewAll/AdminMyTeamIncidentViewAll.jsx': 'users',
    'src/pages/Admin/AdminAllIncidents/AdminAllIncidents.jsx': 'users',
    'src/pages/SuperAdmin/SuperAdminAllIncident/SuperAdminAllIncident.jsx': '(users || [])',
    'src/pages/User/UserMyTeamIncidentViewAll/UserMyTeamIncidentViewAll.jsx': 'users',
    'src/pages/Technician/TechnicianAllTeam/TechnicianAllTeam.jsx': 'allUsers',
    'src/components/IncidentHistory/IncidentHistory.jsx': 'users',
    'src/pages/Admin/AdminViewIncident/AdminViewIncident.jsx': 'users',
    'src/pages/Technician/TechnicianMyAssignedIncidents/TechnicianMyAssignedIncidents.jsx': 'allUsers'
}

for path, var_name in files.items():
    if not os.path.exists(path):
        continue
    with open(path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    replacement = f"""const getUserName = (serviceNumber) => {{
    if (!serviceNumber || String(serviceNumber).trim() === '') return 'Unassigned';
    if (!Array.isArray({var_name})) return serviceNumber;
    const foundUser = {var_name}.find(
      (user) => String(user.service_number) === String(serviceNumber) || String(user.serviceNum) === String(serviceNumber)
    );
    return foundUser ? (foundUser.display_name || foundUser.user_name || foundUser.name || serviceNumber) : serviceNumber;
  }};"""
    
    new_content = re.sub(r'const getUserName = \(serviceNumber\) => \{[\s\S]*?return[\s\S]*?\};', replacement, content)
    
    with open(path, 'w', encoding='utf-8') as f:
        f.write(new_content)
    print(f'Fixed {path}')
